DEVELOPER.md

```diff
@@ -0,0 +1,263 @@
+# EthereumJS - Developer Docs
+
+This guide provides an overview of the monorepo, development tools used, shared configuration and additionally covers some advanced topics.
+
+It is intended to be both an entrypoint for external contributors as well as a reference point for team members.
+
+## Contents
+
+- [Monorepo](#monorepo)
+  - [Structure](#structure)
+  - [Workflow](#workflow)
+- [Development Tools](#development-tools)
+  - [TypeScript](#typescript)
+  - [Linting](#linting)
+  - [Spellcheck](#spellcheck)
+  - [Testing](#testing)
+  - [Documentation](#documentation)V
+- [Advanced Topics](#advanced-topics)
+  - [E2E Testing](#e2e-testing)
+  - [Cross-Package Development](#cross-package-development)
+- [Additional Docs](#additional-docs)
+  - [VM](#vm)
+  - [Client](#client)
+
+## Monorepo
+
+### Structure
+
+The EthereumJS project uses [npm workspaces](https://docs.npmjs.com/cli/v7/using-npm/workspaces) to manage all the packages in our monorepo and link packages together.
+
+#### Key Directories
+
+- `/packages` - Contains all EthereumJS packages
+- `/config` - Shared configuration files and scripts
+- `/ethereum-tests` - Git submodule with Ethereum test vectors
+
+### Scripts
+
+The `./config/cli` directory contains helper scripts referenced in package.json files:
+
+- `coverage.sh` - Runs test coverage
+- `lint.sh` - Checks code style
+- `lint-fix.sh` - Automatically fixes code style issues
+- `ts-build.sh` - Builds TypeScript for production
+- `ts-compile.sh` - Compiles TypeScript for development
+
+### Workflow
+
+#### Common Commands
+
+- **Clean the workspace**: `npm run clean` - Removes build artifacts and node_modules
+- **Lint code**: `npm run lint --workspaces` - Check code style with ESLint v9 and Biome
+- **Fix linting issues**: `npm run lint:fix --workspaces` - Automatically fix style issues
+- **Build all packages**: `npm run build --workspaces` - Build all packages in the monorepo
+- **Build documentation**: `npm run docs:build --workspaces` - Generate documentation for all packages
```

Comment:
Do we have to add `--workspaces` from the monorepo root dir? I think we did not have to do that for these specific commands listed in this section.

Reply:
For `lint`, `lint:fix`, and `docs:build` scripts, no need to append `--workspaces`.  Doing `--workspaces` will actually make the linter run even slower.